app/_src/mesh/guides/deploy-kong-mesh-using-terraform-and-konnect.md

```diff
@@ -0,0 +1,606 @@
+mesh     = konnect_mesh.mesh1.name
+depends_on = [konnect_mesh.mesh1]
+labels = {
+    "kuma.io/mesh" = konnect_mesh.mesh1.name
+}
+```
+
+{% warning %}
+Autogenerated labels like "kuma.io/mesh", "kuma.io/origin" etc. have to be manually added to the resources.
+This limitation will be removed in the GA release.
+{% endwarning %}
+
+The resulting resource looks like this:
+
+```hcl
+resource "konnect_mesh_traffic_permission" "allow_all" {
+    provider = konnect-beta
+    cp_id    = konnect_mesh_control_plane.my_meshcontrolplane.id
+    mesh     = konnect_mesh.mesh1.name
+    depends_on = [konnect_mesh.mesh1]
+    labels = {
+        "kuma.io/mesh" = konnect_mesh.mesh1.name
+    }
+    type = "MeshTrafficPermission"
+    name = "allow-all"
+    spec = {
+        from = [
+            {
+                target_ref = {
+                    kind = "Mesh"
+                }
+                default = {
+                    action = "Allow"
+                }
+            }
+        ]
+    }
+}
+```
+
+## Impact of Renaming Resources
+
+Certain properties (like Mesh name, policy name, etc.) are used as identifiers and changing them will result in a new resource being created and all dependant resources being recreated.
+
+So changing mesh name to `another-name`
+
+```hcl
+resource "konnect_mesh" "mesh1" {
+  # ...
+  name = "another-name"
+  # ...
+}
+```
+
+Will result in forced replacement of both `mesh` and `mesh_access_log` resources:
+
+```
+    # konnect_mesh.mesh1 must be replaced
+-/+ resource "konnect_mesh" "mesh1" {
+      ~ name                           = "mesh1" -> "another-name" # forces replacement
+        # (4 unchanged attributes hidden)
+    }
+
+  # konnect_mesh_access_log.konnect_mesh_traffic_permission must be replaced
+-/+ resource "konnect_mesh_traffic_permission" "allow_all" {
+      ~ creation_time     = "2025-03-13T09:53:00.606442Z" -> (known after apply)
+      ~ labels            = {
+          ~ "kuma.io/mesh" = "mesh1" -> "another-name"
+        }
+      ~ mesh              = "mesh1" -> "another-name" # forces replacement
+```
+
+## Deploying a Kubernetes Zone
+
+To deploy a Kubernetes zone you can use any Kubernetes services, in this guide we will use `k3d` to create a local Kubernetes cluster.
+
+Create a new k3d cluster:
+```bash
+k3d cluster create tfmink
+```
+
+Write `tfmink` cluster configuration:
+```bash
+export KUBECONFIG=$(k3d kubeconfig write tfmink)
+```
+
+Update the `main.tf` file to include the time, kubernetes and helm provider:
+
+```hcl
+terraform {
+  required_providers {
+    konnect = {
+      source = "kong/konnect"
+    }
+    konnect-beta = {
+      source  = "kong/konnect-beta"
+      version = "{{ TF_BETA_VERSION }}"
+    }
+    time = {
+      source  = "hashicorp/time"
+      version = "{{ TF_TIME_VERSION }}"
+    }
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = "{{ TF_KUBERNETES_VERSION }}"
+    }
+    helm = {
```

Comment:
```
╷
│ Error: Kubernetes cluster unreachable: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
│
│   with helm_release.kong_mesh,
│   on a.tf line 117, in resource "helm_release" "kong_mesh":
│  117: resource "helm_release" "kong_mesh" {
│
```

I think we're missing a `provider "helm"` line to configure kubeconfig to use the var

Reply:
you're right on both comments - I think it's fixed now :)